batteryparameter_check.c

```diff
@@ -0,0 +1,73 @@
+#include "batteryparameters.h"
+#include "battery_range_tolerance_handling.h"
+#include <stdio.h>
+
+
+//parameter ranges
+const float min_temp = 0;
+const float max_temp = 45;
+const float min_soc = 20;
+const float max_soc = 80;
+const float min_CR = 0.0;
+const float max_CR = 0.8;
+
+//tolerance levels
+ float min_tol_temp =0 ;
+ float max_tol_temp =0 ;
+ float min_tol_soc =0 ;
+ float max_tol_soc =0;
+ float min_tol_CR =0 ;
+ float max_tol_CR =0;
+
+
+int tempIsok (float temperature)
+{
+     int rangecheck_temp = 0 ;
+    
+     printf("battery temperature:\n");  
+     rangecheck_temp = rangecheck (temperature,max_temp,min_temp);     
+    
+     calculate_tolerance (max_temp,min_temp,&max_tol_temp,&min_tol_temp);
```

Comment:
This function is very similar to the next two. Adding a parameter will result in copy-pasting this function once more.
Can you remove the duplication?